relation_engine_server/utils/ensure_specs.py

```diff
@@ -228,6 +228,25 @@ def excise_namespace(analyzer_name: str) -> str:
     return analyzer_name.split("::")[-1]
 
 
+def is_obj_subset_rec(
+    l: Union[dict, list, float, str, int],
+    r: Union[dict, list, float, str, int],
+):
+    """
+    Compare two JSON objects, to see if, essentially, l <= r
+    If comparing dicts, recursively compare
+    If comparing lists, shallowly compare. For now, YAGN more
+    """
+    if isinstance(l, dict) and isinstance(r, dict):
+        return all(
+            [k in r.keys() and is_obj_subset_rec(l[k], r[k]) for k in l.keys()]
+        )  # ignore: typing
+    elif isinstance(l, list) and isinstance(r, list):
+        return all([le in r for le in l])
+    else:
+        return l == r  # noqa: E741
+
+
```

Comment:
My first guess on how to compare the local/server specs was just `local.items() <= server.items()` with the assumption that ArangoDB would only add default parameters. Kind of worked, but ended up working better with recursive subset checks ..